src/solvers/quasi_Newton.jl

```diff
@@ -77,15 +79,18 @@ function QuasiNewtonState(
     VTM<:AbstractVectorTransportMethod,
 }
     sk_init = zero_vector(M, p)
-    return QuasiNewtonState{P,typeof(sk_init),D,SC,S,RM,VTM}(
+    stepsize_storage = StepsizeStorage(M, stepsize; p_init=p, X_init=initial_vector)
```

Comment:
...for example why does this have to be done here? If I pass a stepsize into the low level function I do not want it „decorated“ that should happen before. I would assume that the new Linesearches can do that themselves?
This also seems the only change with the new field

Let‘s keep it simple here and allow all step sizes. So I would prefer if the LineSearch.jl Stepsiyzes “wrap themselves” into storage if they need to. I would do the same with the Manopt ones (a constant stepsize never need _any_ storage for example).

Also that way one can share the storage later. If it is always only generated here in that is never possible.

Reply:
I can only create storage when I have manifold, point and vector. I don't have them earlier and forcing users to supply them to create a stepsize would be awkward. Anyway, this goes in the general direction of sharing storage much more than having storage completely separate: search state creation "asking" different things it relies on what kind of storage they need and merging them into one.

Reply:
We provide them to all other „small“ structures (debug/record) and they all just recently got a manifold.

But doing the storage creation here and leaving all old ones as „pass through“ is completely the opposite of the current modular design _and_ will be a lot of work when trying to work further on storage, since then we (or I) have to fix two completely different modes of how storage is handled.

And this does not go into another direction – storage is a decorator – ok, so is debug (having a storage internally), but it is still separate storage (not in state but in the stepsize now), so this is kind of the two disadvantages of bot approaches combined: It is modular storage (not central) that _never_ can be shared. Then you Can also just introduce 3 fields in the Stesize.

And ArmijoStepsize for example _has_ a manifold, so that is also not strange.

Reply:
and you are correct my storage design is not the best of choices. But introducing now in different step sizes and different solvers new ways to store stuff makes the whole package a little too chaotic, unless all models used are in detail documented. And I don‘t want to document 2-3 (or 4 or 5 or how many we prototype this way then) motivations on how to do storage.

I would prefer we follow the current way of storage here as well until we have a nice (maybe state-central) way of doing so.

Reply:
I'm a bit tired fighting with the storage design and I most likely won't need Manopt-LineSearches compatibility so I think I won't be finishing this for some time.

Reply:
I understand your position on my design, and I agree that storage should be consistent. LineSearches.jl didn't improve the performance as I expected it would, sadly.

Another option for finishing this PR is removing the initial step size selection part. What would remain wouldn't require any storage so there would be no design issues and you could think later about a way to extend initial step size choice to existing Manopt line searches as well.

Reply:
Ah that also sounds like an idea to finish this, yes. That would be less work at least. Maybe I'll go for that.then when I find time. You are right, its nice to have these new line searches available here.


And sure I can understand your tendency to a new storage, that is why we already have an issue for that :) just that this time I want to spent more thinking about the design first.